### PR TITLE
Refactored to add subject to role creation.

### PR DESCRIPTION
**Pull request by danielfoglio**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,6 @@ override.tf.json
 
 # Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
 # example: *tfplan*
+
+# IntelliJ
+.idea
```
Comment (seanmorton) [inline .gitignore:32]: We can keep this but wanted to let you know about [global gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) if you didn't know already, it's a nice way to ignore your editor files without having to do so in each new repository. Here's how it's setup on my config to ignore vim files:
```bash
# ~/.gitignore                                                                                                                   
*.sw?
.ctags
tags
```
and enable it your git config file, or you can also run `git config --global core.excludesfile ~/.gitignore`
```
# ~/.gitconfig
[core]
  excludesfile = /Users/{your_username}/.gitignore
```

File: modules/aws-oidc-github/variables.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  description = "GitHub URL to perform TLS verification against."
+}
+
+variable "aud_value" {
+  type        = string
+  default     = "sts.amazonaws.com"
+  description = "GitHub Aud"
+}
+
+variable "match_field" {
```
Comment (seanmorton) [inline modules/aws-oidc-github/variables.tf]: Does this makes sense as a variable? I think it will always be `sub` right?

File: .gitignore (hunk 1 above)
Comment (danielfoglio) [reply .gitignore:32]: Oh nice! Yah I did not know about the global gitignore!

File: modules/aws-oidc-github/variables.tf (hunk 2 above)
Comment (danielfoglio) [reply modules/aws-oidc-github/variables.tf]: Yah, I don't think we need this! I will go ahead and remove it

**Review by seanmorton — APPROVED** [review]

**Pull request merged**